DllImportGenerator/DllImportGenerator/DllImportGenerator.cs

```diff
@@ -24,17 +24,12 @@ public class DllImportGenerator : ISourceGenerator
 
         public void Execute(GeneratorExecutionContext context)
         {
-            var synRec = context.SyntaxReceiver as SyntaxReceiver;
-            if (synRec is null || !synRec.Methods.Any())
+            if (context.SyntaxContextReceiver is not SyntaxContextReceiver synRec
```

Comment:
I guess this is fine. I would like to say though that having a side effect of a negated statement be a declaration that one can use is confusing and in my opinion not as clear as what it was previously. If this is preferred then okay, but if we are going for clarity this syntax is  confusing at compared to what it was previously.

Reply:
Personally I like it just so that I don't need to have the mental model of what `as` does on failure in addition to the negative check.